pages/reports/index.js

```diff
@@ -45,11 +38,8 @@ export default function ReportsPage({ transactions }) {
         filter={filter}
         setFilter={setFilter}
```

Comment:
you should not drill the setter function to another component. This makes you lose control how the useState will be changed. It' s better to define a function e.g. handleChangeCategory(), which does exactly only this change. Then drill the handleChangeCategory function to wherever you need it.